manifests/config/passenger/fragment.pp

```diff
@@ -7,23 +7,25 @@
 #  $ssl_content:: content of the ssl virtual host fragment
 #
 define foreman::config::passenger::fragment(
-  $content='',
-  $ssl_content='',
+  $content=undef,
+  $ssl_content=undef,
 ) {
   require foreman::config::passenger
 
-  file { "${apache::confd_dir}/05-foreman.d/${name}.conf":
-    ensure  => present,
-    content => $content,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0644',
-    notify  => Class['apache::service'],
+  if $content {
```

Comment:
It's most like OK, but since it's configurable I'm leaning to explicit. Maybe determine the ensure status based on ```$content```? I think that's cleaner than an else block. Also, puppet 4 will still write an empty file since ```''``` will evaluate to true.